packages/terra-table/src/Table.jsx

```diff
@@ -0,0 +1,326 @@
+   */
+  pinnedColumns: PropTypes.arrayOf(columnShape),
+
+  /**
+   * Data for overflow columns. Overflow columns are rendered in the table's horizontal overflow.
+   * Columns will be presented in the order given.
+   */
+  overflowColumns: PropTypes.arrayOf(columnShape),
+
+  /**
+   * Number indicating the default column width in px. This value will be used if no overriding width value is provided on a per-column basis.
+   */
+  defaultColumnWidth: PropTypes.number,
+
+  /**
+   * String that specifies the column height. Any valid CSS height value is accepted.
+   */
+  columnHeaderHeight: PropTypes.string,
+
+  /**
+   * String that specifies the height for the rows in the table. Any valid CSS value is accepted.
+   */
+  rowHeight: PropTypes.string,
+
+  /**
+   * Number indicating the index of the column that represents row header. Index is 0 based and cannot exceed one less than the number of columns in the table.
+   */
+  rowHeaderIndex: validateRowHeaderIndex,
+
+  /**
+   * Function that is called when a resizable column is resized. Parameters:
+   * @param {string} columnId columnId
+   * @param {string} requestedWidth requestedWidth
+   */
+  onColumnResize: PropTypes.func,
+
+  /**
+   * Callback function that is called when a selectable cell is selected. Parameters:
+   * @param {string} rowId rowId
+   * @param {string} columnId columnId
+   */
+  onCellSelect: PropTypes.func,
+
+  /**
+   * Callback function that is called when a selectable column is selected. Parameters:
+   *  @param {string} columnId columnId
+   */
+  onColumnSelect: PropTypes.func,
+
+  /**
+   * @private
+   * The intl object containing translations. This is retrieved from the context automatically by injectIntl.
+   */
+  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
+};
+
+const defaultProps = {
+  rowHeaderIndex: 0,
+  defaultColumnWidth: 200,
+  columnHeaderHeight: '2.5rem',
+  rowHeight: '2.5rem',
+  pinnedColumns: [],
+  overflowColumns: [],
+};
+
+const defaultColumnMinimumWidth = 60;
+const defaultColumnMaximumWidth = 300;
+
+function Table(props) {
+  const {
+    id,
+    ariaLabelledBy,
+    ariaLabel,
+    rows,
+    pinnedColumns,
+    overflowColumns,
+    onColumnResize,
+    defaultColumnWidth,
+    columnHeaderHeight,
+    rowHeight,
+    onColumnSelect,
+    onCellSelect,
+    rowHeaderIndex,
+    intl,
+  } = props;
+
+  const [pinnedColumnOffsets, setPinnedColumnOffsets] = useState([]);
+
+  // Initialize column width properties
+  const initializeColumn = (column) => ({
+    ...column,
+    width: column.width || defaultColumnWidth,
+    minimumWidth: column.minimumWidth || defaultColumnMinimumWidth,
+    maximumWidth: column.maximumWidth || defaultColumnMaximumWidth,
+  });
+
+  const displayedColumns = [
+    ...pinnedColumns,
+    ...overflowColumns,
+  ];
+  const [renderedColumns, setRenderedColumns] = useState(displayedColumns.map((column) => initializeColumn(column)));
```

Comment:
Why do we need both displayed columns and rendered columns?  I would expect that we need only one like in the DataGrid implementation.  This is confusing have so many variables for our columns.

Reply:
I attempted to re-factor this but it seems to be a non-trivial change so I reverted this back to how it used to be in DataGrid. I agree though that this is not an ideal way to manage the table columns.